client/client.go

```diff
@@ -48,23 +48,53 @@ var (
 // based implementation MUST return this error when it receives a 404 StatusCode.
 type Fetcher func(ctx context.Context, path string) ([]byte, error)
 
+// CheckpointFetcher is the signature of a function which can retrieve the latest
```

Comment:
```suggestion
// CheckpointFetcherFunc is the signature of a function which can retrieve the latest
```

But maybe `GetCheckpointFn` or `GetCheckpointFunc` is better? That leaves the `Fetcher` terminology still mostly pointing at path-based approaches.